packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -45,72 +63,235 @@ import {
   const hasLocationSettingSort =
     context.params.query &&
     context.params.query.$sort &&
     Object.keys(context.params.query.$sort).some((item) => locationSettingSorts.includes(item))
 
-  if (hasLocationSettingSort) {
-    const { dispatch } = context
-
+  if (hasLocationSettingSort && context.params.query && context.params.query.$sort) {
     for (const sort of Object.keys(context.params.query.$sort)) {
       if (locationSettingSorts.includes(sort)) {
-        const data = dispatch.data ? dispatch.data : dispatch
+        const currentSort = context.params.query.$sort[sort]
+        delete context.params.query.$sort[sort]
 
-        data.sort((a, b) => {
-          let fa = a.locationSetting[sort],
-            fb = b.locationSetting[sort]
+        const query = context.service.createQuery(context.params)
 
-          if (typeof fa === 'string') {
-            fa = fa.toLowerCase()
-            fb = fb.toLowerCase()
-          }
+        query.join(locationSettingPath, `${locationSettingPath}.locationId`, `${locationPath}.id`)
+        query.orderBy(`${locationSettingPath}.${sort}`, currentSort === 1 ? 'asc' : 'desc')
+        query.select(`${locationPath}.*`)
+
+        context.params.knex = query
+      }
+    }
+  }
+}
+
+const addSearchParams = async (context: HookContext<LocationService>) => {
+  const { adminnedLocations, search } = context.params.query || {}
 
-          if (fa < fb) {
-            return -1
+  if (adminnedLocations && search) {
+    context.params.query = {
+      ...context.params.query,
+      $or: [
+        {
+          name: {
+            $like: `%${search}%`
           }
-          if (fa > fb) {
-            return 1
+        },
+        {
+          sceneId: {
+            $like: `%${search}%`
           }
-          return 0
+        }
+      ]
+    }
+  }
+  if (context.params.query?.adminnedLocations) delete context.params.query.adminnedLocations
+  if (context.params.query?.search || context.params.query?.search === '') delete context.params.query.search
+}
+
+const insertLocationData = async (context: HookContext<LocationService>) => {
+  if (!context.data || context.method !== 'create') {
+    throw new BadRequest(`${context.path} service only works for data in ${context.method}`)
+  }
+
+  const data: LocationData[] = Array.isArray(context.data) ? context.data : [context.data]
+
+  const trx = await (context.app.get('knexClient') as Knex).transaction()
+
+  const result: LocationType[] = []
+
+  for (const item of data) {
+    try {
+      const selfUser = context.params?.user
+
+      if (item.isLobby) {
+        await context.makeLobby(trx, selfUser)
+      }
+
+      item.slugifiedName = slugify(item.name, { lower: true })
+
+      const insertData = JSON.parse(JSON.stringify(item))
+      delete insertData.locationSetting
+      delete insertData.locationAdmin
+
+      await trx.from<LocationDatabaseType>(locationPath).insert(insertData)
```

Comment:
Just wandering if we can split this whole function into before and afters and use knex transation like: https://feathersjs.com/api/databases/knex.html#transactions

Same goes for `updateLocation` hook